Grbl_Esp32/src/Protocol.cpp

```diff
@@ -167,18 +167,49 @@ void protocol_main_loop() {
     int c;
     for (;;) {
         auto sdcard = config->_sdCard;
-        if (sdcard->_ready_next) {
+
+        // _readyNext means a line *might* be available from the _SDCard
+        // and implies SDCard::State == Busy (which is the same value as BusyPrinting).
+
+        if (sdcard->_readyNext) {
             char fileLine[255];
-            if (sdcard->readFileLine(fileLine, 255)) {
-                sdcard->_ready_next = false;
+
+            sdcard->_readyNext = false;
+                // _readyNext is invariantly set to false here.
+                // either readFileLine() will succeed, in which case
+                // it is up to report_status_message() to reset it,
+                // or we are done with EOF or an error and we want to
+                // leave it as false.
+
+            SDCard::ReadResult_t rslt = sdcard->readFileLine(fileLine, 255);
+
+            #define DEBUG_PROTOCOL_READFILELINE 0
+
+            if (rslt == SDCard::ReadResult_t::OK) {
```

Comment:
We prefer switch() to if () else if () else if() when the test is on the same variable, especially if it is an enumerated type.  The compiler can catch errors like missing cases, and the intent of the code is clearer; the reader does not have to look at all the if clauses to determine that it is handling all the possible values of the variable, with no other conditions thrown into the mix.